### DISCUSSION TITLE
Push notification/information, from the internet

### DISCUSSION THREAD

**Discussion by OttoMan28** [micropython/micropython#12122] [General]
Greetings,

I'm a little out of my comfort zone here, so perhaps you could guide me in the right direction.

I need to establish communication with a handful of MP's devices. I aim to distribute information from a central server (on the internet) to these MP devices.
To avoid the hassle of configuring the firewall on the local network, I need the MP device to initiate the communication. Could you suggest any methodologies or libraries that might be of help?

Thanks,
Otto

**Comment by karfas** [comment]
Create some kind of web service, static html pages or the like on the internet side and use (u)request.py to connect.
Most firewalls will not block outgoing requests to ports 80/httpd or 443/https.

**Reply by OttoMan28** [reply]
But that will result in a pull/polling configuration, what I'm looking for is a push configuration.
I dont what to have the clients polling the server all the time.

**Reply by karfas** [reply]
Looks like I missed that in your original requirements.
MQTT or other subscribe-based message brokers will fit.

**Reply by OttoMan28** [reply]
I think MQTT will fit the bill, but other brokers could you recommend ?

**Comment by sosi-deadeye** [comment]
You could use `WebSocket`, which is a kind of persistent `http(s)` connection. The server could send the connected clients new frames if an event happens. AFIK `https` is required on the internet for Websockets, but this counts only for web browser.

On both, server- and client-side you should use `asyncio`. The most time, the program is just waiting for IO.

Another alternative is a password protected `mqtt` server on the server. From micropython-side you connect and listen for new messages with topics, which are subscribed.

**Reply by OttoMan28** [reply]
MQTT of cause :-) i have used it so many times for internal use, but i didn't think of it here.
I was over-thinking the solution. Here i might get a small dataset 2-4 times a day. And prehaps sendt a status once a day.
Thanks for straighten out my head :-)
